### ISSUE TITLE
Memory exceed issue after the first round

### ISSUE THREAD

**Issue by MikeHuYimin** [apache/incubator-mxnet#2731]
Currently I am working on a huge data set. The thing is that if I just put the whole training data in mx.Create.Feedforward, the memory will soon be filled up and fail to run. A easy example is like that.
`data <- mx.symbol.Variable("data")`
`fc1 <- mx.symbol.FullyConnected(data, num_hidden = 3000)`
`fc2 <- mx.symbol.FullyConnected(fc1, num_hidden = 10000)`
`out <- mx.symbol.LinearRegressionOutput(data = fc2)`
`net.dl <- mx.model.FeedForward.create(
  X = input,
  y = truth,
  symbol = out,
  ctx = mx.cpu(),
  num.round = 10,
  learning.rate = 0.001,
  momentum = 0.9,
  eval.metric = mx.metric.rmse,
  array.batch.size = 10,
  array.layout = "colmajor"
)`
The weird thing is that it can run the first round and get the rmse result like:
[1] Train-rmse=2.81730335008859
Then the memory filled up and my job was killed. Why crush after the first round? Will the memory requirement become larger in the deeper round?  Actually I am working on a 512GB mem compute node on the cluster, my data is about 6GB (10000*100000 matrix). I suppose it won't take so much memory.(If I change the hidden nodes to 300 it's ok but that's not the thing I want)

**Comment by phunterlau** [comment]
This issue is closed due to lack of activity in the last 90 days. Feel free to reopen if this is still an active issue. Thanks!
